x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/SecuritySettings.java

```diff
@@ -19,9 +19,10 @@ public static Settings addTransportSettings(final Settings settings) {
         final Settings.Builder builder = Settings.builder();
         if (NetworkModule.TRANSPORT_TYPE_SETTING.exists(settings)) {
             final String transportType = NetworkModule.TRANSPORT_TYPE_SETTING.get(settings);
-            if (SecurityField.NAME4.equals(transportType) == false) {
+            if (SecurityField.NAME4.equals(transportType) == false && SecurityField.NIO.equals(transportType) == false ) {
                 throw new IllegalArgumentException("transport type setting [" + NetworkModule.TRANSPORT_TYPE_KEY
-                        + "] must be [" + SecurityField.NAME4 + "] but is [" + transportType + "]");
+                        + "] must be [" + SecurityField.NAME4 + "] or [" + SecurityField.NIO + "]" +" but is ["
+                    + transportType + "]");
```

Comment:
I'm going to be pedantic about the inconsistent indenting of this multi-line string.